### ISSUE TITLE
Pre-trained model

### ISSUE THREAD

**Issue by EmreOzkose** [yuval-reshef/StreamVC#1]
Hi, congrats to reproducing this work.

I have a few questions.
1. Do you have a plan to share pre-trained models?
2. Is the model capable of converting between different languages?
3. How much audio data do we need to convert like 5 minutes, 10 minutes, etc..? 

Thank you in advance :)

**Comment by yuval-reshef** [comment]
Hi.
1. I currently don't have the resources to train the model properly. This can become pretty expensive, since the hyper parameters were not shared in the paper. We did however test it and saw it can be trained.
If someone is willing to put in the work and money, I'll be happy to share their per-trained model instance in this repo. Otherwise, I might do it in the future but can't commit on a timeline.
3. Without a fully trained model, I can't answer this question.
4. 5-10 seconds for target speech is usually enough going by the paper's results page (https://google-research.github.io/seanet/stream_vc/)

**Comment by hermanseu** [comment]
Hi @yuval-reshef 
What is the accuracy of your content-encoder model？I can just get 47%, based on the context-encoder,  I tained the left model, and get extremely poor performance.  Can you share the  per-trained model to have a try?

**Comment by RamakrishnaChaitanya** [comment]
Hi @hermanseu,

May i know the hyper parameters that you have used during the training? I tried training using the existing code as is, but, the execution is getting stopped after 1380 steps. (FYI, i used num_epochs = 1, just to check the execution status). After 1380 steps, the obtained accuracy was around 3.13%. 

I'm using the following splits. 
![image](https://github.com/user-attachments/assets/8483dcda-6e50-4f81-822e-6c4506d01253)

**Comment by RamakrishnaChaitanya** [comment]
Hi @hermanseu ,

May i know the hyper parameters that you used to achieve 47 % accuracy on the content encoder. I'm getting only 3% - 4% accuracy even after 10 epochs. It will be helpful for me if you can share your hyper parameters. Thanks in advance. 

FYI, i used the code as it is.

**Comment by hermanseu** [comment]
@RamakrishnaChaitanya 

Below is my train command. Buy the way, I still haven't got a successful model.
accelerate launch --config_file ~/accelerate_config_gpu_0123_fp32.yaml ./train.py  \
     --module-to-train content-encoder \
     --checkpoint-path /data/212_data4/streamvc_vq_hidden \
     --train-data-path /data/212_data4/train_data/train \
     --val-data-path /data/212_data4/train_data/val \
     --kmeans-ckpt-path /work/02_PyProjects/62_vc/kmeans100-50f36a95.pt \
     --hubert-discrete-ckpt-path /work/02_PyProjects/62_vc/hubert-discrete-96b248c5.pt \
     --lr 4e-3 \
     --scheduler-gamma 0.8 \
     --encoder-dropout 0.2 \
     --num-epochs 200 \
     --scheduler-step 5 \
     --accuracy-interval 1000 \
     --model-checkpoint-interval 5000 \
     --log-interval 100 \
     --log-labels-interval 100

**Comment by RamakrishnaChaitanya** [comment]
Hi @hermanseu 

Thanks for your input. I tried training the content encoder model with the similar set of the hyper parameters as you inputted (mentioned below), however, i still don't see any improvement in the training loss or validation accuracy. Even after 10 epoch's, the accuracy remained constant at ~ 3 %. 

accelerate launch train.py 
--run-name Hyper_params_v1 
--module-to-train content-encoder
 --lr 4e-3
 --scheduler-gamma 0.8
 --encoder-dropout 0.2
 --num-epochs 500
 --scheduler-step 5
 --accuracy-interval 1000
 --model-checkpoint-interval 1000
 --log-interval 100
 --log-labels-interval 100

I'm using the "train.clean.100" split from [LibriTTS_r](https://huggingface.co/datasets/mythicinfinity/libritts_r?row=1) dataset for training the model and using "dev.clean" split as the validation dataset Are you also using the same datasets in your training?

**Comment by RamakrishnaChaitanya** [comment]
Thanks for your input @hermanseu
